device/iot-device-samples/device-reconnection-sample/src/main/java/samples/com/microsoft/azure/sdk/iot/DeviceClientManagerSample.java

```diff
@@ -30,38 +26,75 @@
      */
     public static void main(String[] args)
             throws URISyntaxException, IOException {
-        log.debug("Starting the sample...");
-        log.debug("Using communication protocol: {}", protocol.name());
 
-        DeviceClient client = new DeviceClient(deviceConnectionString, protocol);
-        log.debug("Successfully created an IoT Hub client.");
+        SampleParameters params = new SampleParameters(args);
+
+        log.info("Starting...");
+        log.info("Setup parameters...");
+
+        String argDeviceConnectionString = (params.getConnectionStrings())[0];
+
+        IotHubClientProtocol argProtocol;
+        String protocol = params.getTransport().toLowerCase();
+        switch (protocol)
```

Comment:
Consider holding these pairs in a map.

Reply:
There are pros and cons with map vs switch. For our small set of options and use-it-once use case, there isn't significant difference in terms of performance or readability testing both. I will have a separate commit for the map version so in case we want to you either one, there is a specific commit for the clean file for such.

Reply:
No problem; as I started my sentence with the word "consider", I meant it to be optional.